### DISCUSSION TITLE
Contributing on Mac

### DISCUSSION THREAD

**Discussion by ucodery** [starship/starship#4836] [Q&A] [answered]
Hi, I am trying to build a checkout of Starship locally so that I can maybe contribute some PRs, but I can't get current `master` to build and would love some help debugging. The failure from running `cargo build` looks something like
```
cargo:warning=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/QuartzCore.framework/Headers/CABase.h:16:10: fatal error: could not build module 'Core
Foundation'
  cargo:warning=#include <CoreFoundation/CoreFoundation.h>
  cargo:warning= ~~~~~~~~^
  cargo:warning=In file included from objc/notify.m:1:
  cargo:warning=objc/notify.h:1:9: fatal error: could not build module 'Cocoa'
  cargo:warning=#import <Cocoa/Cocoa.h>
  cargo:warning= ~~~~~~~^
  cargo:warning=23 errors generated.
  exit status: 1

  --- stderr


  error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-arch" "x86_64" "-Wall" "-Wextra" "-fmodules" "-Wno-depr
ecated-declarations" "-mmacos-version-min=10.8" "-o" "/Users/jeremyp/git/starship/target/debug/build/mac-notification-sys-a49030061b8e8737/out/objc/notify.o" "-c" "objc/notify.m" w
ith args "cc" did not execute successfully (status code exit status: 1).
```
[this seems to be a known issue upstream](https://github.com/h4llow3En/mac-notification-sys/issues/28) but Starship is obviously still building as I am using it on my Mac, and the latest CI appears to build on macos (although not pass tests).

My setup is 
* Apple clang version 14.0.0
  * I have also tried using my brew llvm 15.0.7
* rustc 1.65.0
* xcode-select 2395
* macos 12.6.1

**Comment by davidkna** [comment, marked as answer]
Have you tried using Xcode instead of the CLT? According to the issue, Xcode could work. You can also build starship without that library by adding the `--no-default-features` flag.
